### PR TITLE
feat(components): add  `columns` prop in `BaseVariableColumnGrid`

### PR DESCRIPTION
**Pull request by tiborux**
EX-8135

### CODE REVIEW

**Review by LuisMartinez15 — APPROVED** [review]
👏🏿 good job 👏🏿

File: packages/x-components/src/components/base-variable-column-grid.vue [hunk 1]
```diff
@@ -55,7 +55,26 @@
...
+
+    /**
+     * The number of columns provided by the user.
+     *
+     * @internal
+     */
+    protected providedColumns = 0;
+
+    /**
+     * The number of columns to render in the grid.
```
Comment (CachedaCodes) [inline packages/x-components/src/components/base-variable-column-grid.vue:69]: `columns` and `columnsToRender` have almost the same description.

File: packages/x-components/src/components/base-variable-column-grid.vue [hunk 2]
```diff
@@ -55,7 +55,26 @@
      *
      * @internal
      */
-    protected columns = 0;
+    @Prop({ default: 0 })
+    protected columns!: number;
+
+    /**
+     * The number of columns provided by the user.
```
Comment (CachedaCodes) [inline packages/x-components/src/components/base-variable-column-grid.vue:62]: Can we change to `provided by a user interaction`? For me, it's clearer that way

**Review by CachedaCodes — CHANGES_REQUESTED** [review]
gj!

File: packages/x-components/src/components/base-variable-column-grid.vue [hunk 3]
```diff
@@ -51,11 +51,31 @@
...
-    protected columns = 0;
+    @Prop({ default: 0 })
+    protected columns!: number;
+
+    /**
+     * The number of columns provided by a user interaction.
+     *
+     * @internal
+     */
+    protected providedColumns = 0;
```
Comment (tajespasarela) [inline packages/x-components/src/components/base-variable-column-grid.vue:67]: We are going to need to provide 0 columns too as is for "automatic" columns number. So no provided columns should be null.

```suggestion
    protected providedColumns: number | null = null;
```

File: packages/x-components/src/components/base-variable-column-grid.vue [hunk 4]
```diff
@@ -51,11 +51,31 @@
...
+
+    /**
+     * The number of columns to render in the grid.
+     *
+     * @returns The number of columns.
+     *
+     * @internal
+     */
+    protected get columnsToRender(): number {
+      return this.providedColumns || this.columns;
```
Comment (tajespasarela) [inline packages/x-components/src/components/base-variable-column-grid.vue:77]:
```suggestion
      return this.providedColumns === null?  this.columns: this.providedColumns;
```

**Review by tajespasarela — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Review by tajespasarela — APPROVED** [review]

**Pull request merged**